datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -788,4 +788,18 @@ public Map<String, Map<Integer, Long>> getConsumptionLagForTask(DatastreamTask t
       return taskEntry.getConnectorTask().getKafkaTopicPartitionTracker().getConsumptionLag();
     }
   }
+
+  public List<String> getActiveTasks() {
+    List<DatastreamTask> tasks = new ArrayList<>();
+
+    synchronized (_runningTasks) {
+      synchronized (_tasksPendingStop) {
+        tasks.addAll(_runningTasks.keySet());
+        tasks.addAll(_tasksPendingStop.keySet());
+      }
+    }
+
+    _logger.debug("Found {} active tasks assigned to the connector", tasks.size());
+    return tasks.stream().map(DatastreamTask::getId).collect(Collectors.toList());
```

Comment:
I'm exposing task IDs instead of Task objects on purpose. Exposing Task carries risks of state corruption etc.